CMakeLists.txt

```diff
@@ -204,7 +204,7 @@ include_directories(SYSTEM
   ${OPENGL_INCLUDE_DIR}
   ${PYTHON_INCLUDE_PATH}
 )
-include_directories(src ${catkin_INCLUDE_DIRS})
+include_directories(src SYSTEM ${catkin_INCLUDE_DIRS})
```

Comment:
Sure, the `SYSTEM` directive ends up using `-isystem` as the option to the compiler. The `-isystem` paths are always looked at after `-I`, and if a path is specified with both, the `-I` entry is ignored:

From https://gcc.gnu.org/onlinedocs/cpp/System-Headers.html:
> All directories named by -isystem are searched after all directories named by -I, no matter what their order was on the command line. If the same directory is named by both -I and -isystem, the -I option is ignored.

So that means that when you use it with header paths that may be coming from a workspace, like `~/my_ws/install/include`, and from somewhere else you have an `-I/opt/ros/kinetic/include`, then you can get in a situation where you use the headers from `/opt/ros` rather than the headers from `~/my_ws/install` even when the workspace should take precedence. Also `catkin_INCLUDE_DIRS` is sorted by workspace prefix to avoid a similar problem, so adding these include unevenly may result in unexpected, and hard to debug, behavior when using a workspace.

Also, on OS X, the `isystem` flag is used to control which SDK you're using and it caused problems in the past to have more instances of that, though I think CMake/Apple fixed that at some point.

I think in this case it would ok to use it since we're putting all headers under `SYSTEM`, but because of the subtle issues with `isystem` I don't want to introduce it in case someone adds more headers later without understanding the pitfall. If you don't want to take the time to suppress the warns with the `#pragma`'s, then I'll take care of it. Thanks for the fixes you've provided so far!